elastic/beats
# add_cloud_metadata: env var override for providers

## Proposed commit message

Add support for configuring the add_cloud_metadata providers with an environment variable, `$BEATS_ADD_CLOUD_METADATA_PROVIDERS`. This may be useful when deploying Elastic Agent standalone in a cloud provider (e.g. in a managed Kubernetes cluster), where the cloud provider is known at deployment time.

This is intended as a workaround for https://github.com/elastic/beats/issues/13816 and issues like it.

## Checklist

~~- [ ] I have made corresponding change to the default configuration files~~

## How to test this PR locally

Run Metricbeat enough times on EC2 to trigger https://github.com/elastic/beats/issues/13816, and then check that setting `BEATS_ADD_CLOUD_METADATA_PROVIDERS=ec2` disables the openstack provider and the issue no longer occurs.

## Related issues

Relates to https://github.com/elastic/beats/issues/13816

## Use cases

As a user deploying Elastic Agent standalone on EKS, I want to ensure that documents are enriched with EC2 cloud metadata.

## Screenshots

N/A

## Logs

N/A
